app/server/api/auth_api.py

```diff
@@ -156,18 +156,14 @@ def post(self):
             if tfa_response_oject:
                 tfa_response_oject['auth_token'] = auth_token.decode()
 
+                db.session.commit()  # need this here to commit a created user to the db
```

Comment:
user wasn't being commited to db

Reply:
don't change it now, but I'm wondering if steps like this should actually return 200 codes given the user is going down the correct path.

Reply:
yeah maybe, need to think about it

Reply:
the auth flow, specifically tfa flow, for `/register`, `/login` and `/activate` feels a bit DRY